packages/astro/src/actions/vite-plugin-actions.ts

```diff
@@ -31,7 +32,7 @@ export function vitePluginActionsBuild(
 		name: '@astro/plugin-actions-build',
 
 		applyToEnvironment(environment) {
-			return environment.name === 'ssr';
+			return environment.name === ASTRO_VITE_ENVIRONMENT_NAMES.server;
```

Comment:
in my opinion this should be named `ssr` not `server` to match the Vite-defined name. Soft opinion, open to discussion.